testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestSparkCompatibility.java

```diff
@@ -423,6 +424,35 @@ public void testTrinoShowingSparkCreatedTables()
         onTrino().executeQuery("DROP TABLE " + prestoTableName(trinoTable));
     }
 
+    @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS})
+    public void testPrestoReadingSparkObjectStorageLocationProviderData()
+    {
+        String baseTableName = "test_object_storage_location_provider";
+        String sparkTableName = sparkTableName(baseTableName);
+        String prestoTableName = prestoTableName(baseTableName);
+        String dataPath = "hdfs://hadoop-master:9000/user/hive/warehouse/test_object_storage_location_provider/obj-data";
+
+        String sparkTableDefinition = "CREATE TABLE %s (_string STRING, _bigint BIGINT) USING ICEBERG OPTIONS (" +
+                "'write.object-storage.enabled'=true," +
+                "'write.object-storage.path'='" + dataPath + "')";
+        onSpark().executeQuery(format(sparkTableDefinition, sparkTableName));
+        onSpark().executeQuery(format("INSERT INTO %s VALUES ('a_string', 1000000000000000)", sparkTableName));
```

Comment:
It's important to do an INSERT from Trino here too. That's what runs the changes you made in the `IcebergPageSink`

Reply:
Yeah actually this is a good point, it does not test anything by inserting it using Spark, so I directly changed it to be executed on `onTrino`. 

If we want to test Trino able to read Spark data written by object storage location provider, I think it should be placed in the `testPrestoReadingSparkData` test. But `LocationProvider` is a write side feature, on read side it's just using the file path in the manifest, so I think there is little value for adding that test case.